File: src/TypeScript/TypeScriptLibrary/Library/Measurements/Variables/Variable.ts
import { IObject } from "../../Interfaces/IObject";
import { IValue } from "../../Interfaces/IValue";
import { Performer } from "../../Performer";
import { IMeasurement } from "../Interfaces/IMeasurement";


export class Variable implements IMeasurement, IObject, IValue {
    value: any = new Object();;
    type: any = new Object();
    name: string = "";

    className: string = "Variable";

    types: string[] = ["Variable", "IMeasurement", "IObject", "IValue"];

    performer: Performer = new Performer();

    constructor(name: string, type: any, value: any) {
        this.name = name;
        this.type = type;
        this.value = value;

    }
    getIValue() {
        return this.value;
    }
    setIValue(value: any): void {
        this.value = value;
    }


    getClassName(): string {
        return this.className;
    }
    imlplementsType(type: string): boolean {
        return this.types.indexOf(type) >= 0;
    }
    getName(): string {
        return this.name;
    }

    getMeasurementName(): string {
        return this.name;
    }

    getMeasurementType() {
        return this.type;
    }

    getMeasurementValue() {
        return this.value;
    }



}

File: Java_Generated/Android/measurements/variables/Variable.java
package measurements.variables;
import general_service.interfaces.IValue;
import general_service.interfaces.IValueSetter;
import general_service.interfaces.IValueSetterFactory;
import measurements.interfaces.IDerivation;
import measurements.interfaces.IMeasurement;

import java.lang.reflect.Array;

public class Variable implements IMeasurement,  IValue, IDerivation {
    public Variable(Object type, String name,  IValueSetterFactory factory) {
        this.type = type;
        this.name = name;
        setter = factory.getValueSetter(type);
        value = setter.getDefaultValue();
    }



    @Override
    public Object getIValue() {
        return setter.getValue();
    }

    @Override
    public void setIValue(Object value) {

        setter.setValue(value);
    }

    @Override
    public IMeasurement getDerivation() {
        return derivation;
    }

    @Override
    public void setDerivation(IMeasurement measurement) {
        derivation = measurement;
    }

    @Override
    public String getMeasurementName() {
        return name;
    }

    @Override
    public Object getMeasurementType() {
        return type;
    }

    @Override
    public Object getMeasurementValue() {
        return setter.getValue();
    }

    Object value;

    Object type;

    Array array;

    String name;

    IMeasurement derivation;

    @Override
    public String toString() {
        return setter.toString();
    }

    IValueSetter setter;
    measurements.Performer mPerformer = new measurements.Performer();
}

